client/src/pages/Bookings/Bookings.tsx

```diff
@@ -4,95 +4,77 @@ import useStyles from './useStyles';
 import Calendar from '../../components/Calendar/Calendar';
 import BookingCard from '../../components/BookingCard/BookingCard';
 import 'react-calendar/dist/Calendar.css';
-import { Booking } from '../../interface/Booking';
-import SettingsIcon from '@mui/icons-material/Settings';
+import { Request } from '../../interface/Request';
+import getRequests from '../../helpers/APICalls/getRequests';
+import { useState, useEffect } from 'react';
 
 const boxShadow =
   '0px 0px 1.9px rgba(0, 0, 0, 0.007),0px 0px 4.9px rgba(0, 0, 0, 0.014),0px 0px 9.9px rgba(0, 0, 0, 0.021),0px 0px 20.4px rgba(0, 0, 0, 0.031),0px 0px 56px rgba(0, 0, 0, 0.05)';
 
 export default function Bookings(): JSX.Element {
   const classes = useStyles();
-  // We expect to receive Date objects from the database for all times and dates, and we will build functions to handle conversion in the BookingCard component.
 
-  // This mock will be replaced with database information once implemented
-  const mockBookings: Booking[] = [
-    {
-      user: {
-        name: 'Norma Byers',
-        email: 'example@example.com',
-      },
-      startTime: new Date(Date.parse('2022-01-25T12:00:00.000Z')),
-      endTime: new Date(Date.parse('2022-01-25T14:00:00.000Z')),
-      status: 'accepted',
-    },
-    {
-      user: {
-        name: 'Charles Compton',
-        email: 'example@example.com',
-      },
-      startTime: new Date(Date.parse('2022-01-28T15:00:00.000Z')),
-      endTime: new Date(Date.parse('2022-01-28T18:00:00.000Z')),
-      status: 'accepted',
-    },
-    {
-      user: {
-        name: 'Joan Blakeney',
-        email: 'example@example.com',
-      },
-      startTime: new Date(Date.parse('2022-02-11T08:00:00.000Z')),
-      endTime: new Date(Date.parse('2022-02-11T10:00:00.000Z')),
-      status: 'declined',
-    },
-    {
-      user: {
-        name: 'Michael Carnahan',
-        email: 'example@example.com',
-      },
-      startTime: new Date(Date.parse('2022-01-12T14:00:00.000Z')),
-      endTime: new Date(Date.parse('2022-01-12T16:00:00.000Z')),
-      status: 'accepted',
-    },
-  ];
+  const [mounted, setMounted] = useState(false);
+  const [pastBookings, setPastBookings] = useState<Request[] | undefined>();
+  const [nextBooking, setNextBooking] = useState<Request | undefined>();
+  const [sortedBookings, setSortedBookings] = useState<Request[] | undefined>();
 
-  // return a filtered array AND remove filtered values from the original
-  function getPastBookings(bookingsArray: Booking[]) {
+  useEffect(() => {
+    if (!mounted) {
+      setMounted(true);
+      getRequests().then((res) => {
+        if (res) {
+          const userBookings: Request[] = [];
```

Comment:
Please resort to using the built in array functions in JS unless it doesn't fit our use case